src/main/groovy/com/bmuschko/gradle/docker/tasks/image/Dockerfile.groovy

```diff
@@ -501,20 +501,31 @@ class Dockerfile extends DefaultTask {
     }
 
     interface ItemJoiner {
-        String join(Map map)
+        String join(Map<String, String> map)
     }
 
     static class MultiItemJoiner implements ItemJoiner {
         @Override
-        String join(Map map) {
-            map.inject([]) { result, entry -> result << "\"$entry.key\"=\"$entry.value\"" }.join(' ')
+        String join(Map<String, String> map) {
+            map.inject([]) { result, entry ->
+                // TODO: clarify if assert should be used even in productive code and if this would be a valid one either
+                assert !entry.key.matches('.*(?: |(?:\r{0,1}\n)).*')
```

Comment:
yes, sorry, was on vacation for some days .. As mentioned above, I would only like to escape / encaps whitespaces in quotes and would not like to throw any exception as long we only generate the file as accurate as possible, but are not responsible for validity in any system context, aren't we? So I would replace this assert by a replace as like for the value.

P.S: reading my own comments again .. we need to throw an exception if a blank key is given.